### PR TITLE
Simplify reconciliation

### PR DESCRIPTION
**Pull request by idegtiarenko**

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/cluster/routing/allocation/SingleShardNoReplicasRoutingTests.java [hunk 1]
```diff
@@ -44,6 +44,7 @@
 public class SingleShardNoReplicasRoutingTests extends ESAllocationTestCase {
     private final Logger logger = LogManager.getLogger(SingleShardNoReplicasRoutingTests.class);
 
+    @AwaitsFix(bugUrl = "desired balance")
```
Comment (idegtiarenko) [inline server/src/test/java/org/elasticsearch/cluster/routing/allocation/SingleShardNoReplicasRoutingTests.java:47]: This seems to be the only failure on this branch

**Comment by idegtiarenko** [comment]
@elasticsearchmachine please run elasticsearch-ci/bwc

**Comment by idegtiarenko** [comment]
@elasticsearchmachine please run elasticsearch-ci/part-1

**Review by DaveCTurner — APPROVED** [review]
LGTM, I left only a few tiny nits.

**Comment by idegtiarenko** [comment]
@elasticsearchmachine please run elasticsearch-ci/part-1

**Comment by idegtiarenko** [comment]
@elasticsearchmachine please run elasticsearch-ci/part-2

**Comment by idegtiarenko** [comment]
@elasticsearchmachine please run elasticsearch-ci/part-1

**Pull request merged**